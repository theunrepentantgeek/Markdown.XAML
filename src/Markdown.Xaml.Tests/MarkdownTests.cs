using System.IO;
using System.Reflection;
using System.Threading;
using System.Windows.Markup;
using System.Xml;

using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace Markdown.Xaml.Tests
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class MarkdownTests
    {
        [Test]
        public void Transform_givenTest1_generatesExpectedResult()
        {
            var text = LoadText("Test1.md");
            var markdown = new Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(AsXaml(result));
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void Transform_givenLists_generatesExpectedResult()
        {
            var text = LoadText("Lists.md");
            var markdown = new Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(AsXaml(result));
        }

        [Test]
        public void Transform_givenTables_generatesExpectedResult() {
            var text = LoadText("Tables.md");
            var markdown = new Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(AsXaml(result));
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void Transform_givenHorizontalRules_generatesExpectedResult()
        {
            var text = LoadText("HorizontalRules.md");
            var markdown = new Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(AsXaml(result));
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void Transform_givenLinksInline_generatesExpectedResult()
        {
            var text = LoadText("Links_inline_style.md");
            var markdown = new Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(AsXaml(result));
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void Transform_givenTextStyles_generatesExpectedResult()
        {
            var text = LoadText("Text_style.md");
            var markdown = new Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(AsXaml(result));
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void Transform_givenImages_generatesExpectedResult()
        {
            var text = LoadText("Images.md");
            var markdown = new Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(AsXaml(result));
        }

        private string LoadText(string name)
        {
            using (Stream stream = Assembly.GetExecutingAssembly()
                               .GetManifestResourceStream("Markdown.Xaml.Tests." + name))
            using (StreamReader reader = new StreamReader(stream))
            {
                return reader.ReadToEnd();
            }
        }

        private string AsXaml(object instance)
        {
            using (var writer = new StringWriter())
            {
                var settings = new XmlWriterSettings { Indent = true };
                using (var xmlWriter = XmlWriter.Create(writer, settings))
                {
                    XamlWriter.Save(instance, xmlWriter);
                }

                writer.WriteLine();
                return writer.ToString();
            }
        }
    }
}
