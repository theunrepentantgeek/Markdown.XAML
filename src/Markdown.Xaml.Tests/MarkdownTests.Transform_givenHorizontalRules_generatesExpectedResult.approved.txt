<?xml version="1.0" encoding="utf-16"?>
<FlowDocument PagePadding="0,0,0,0" xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
  <Paragraph>Dashes:</Paragraph>
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph />
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph />
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph />
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph> ---</Paragraph>
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph />
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph />
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph />
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph> - - -</Paragraph>
  <Paragraph>Asterisks:</Paragraph>
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph />
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph />
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph />
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph> <Italic>*</Italic></Paragraph>
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph />
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph />
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph />
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph> * * *</Paragraph>
  <Paragraph>Underscores:</Paragraph>
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph />
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph />
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph />
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph> <Italic>_</Italic></Paragraph>
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph />
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph />
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph />
  <BlockUIContainer>
    <Line X2="1" StrokeThickness="1" />
  </BlockUIContainer>
  <Paragraph> _ _ _</Paragraph>
</FlowDocument>
